src/Sylius/Bundle/ResourceBundle/Controller/ResourceController.php

```diff
@@ -307,179 +379,50 @@ public function disableAction(Request $request)
-    }
-
-    public function moveUpAction(Request $request)
-    {
-        return $this->move($request, 1);
-    }
-
-    public function moveDownAction(Request $request)
-    {
-        return $this->move($request, -1);
-    }
-
-    public function updateStateAction(Request $request, $transition, $graph = null)
+    protected function toggle(Request $request, $enabled)
     {
-        $resource = $this->findOr404($request);
-
-        if (null === $graph) {
-            $graph = $this->stateMachineGraph;
-        }
-
-        $stateMachine = $this->get('sm.factory')->get($resource, $graph);
-        if (!$stateMachine->can($transition)) {
-            throw new NotFoundHttpException(sprintf(
-                'The requested transition %s cannot be applied on the given %s with graph %s.',
-                $transition,
-                $this->config->getResourceName(),
-                $graph
-            ));
-        }
+        $configuration = $this->requestConfigurationFactory->create($this->metadata, $request);
 
-        $stateMachine->apply($transition);
+        $this->isGrantedOr403($configuration, ResourceActions::UPDATE);
 
-        $this->domainManager->update($resource);
+        $resource = $this->findOr404($configuration);
+        $resource->setEnabled($enabled);
 
-        if ($this->config->isApiRequest()) {
-            if ($resource instanceof ResourceEvent) {
-                throw new HttpException($resource->getErrorCode(), $resource->getMessage());
-            }
+        $this->eventDispatcher->dispatchPreEvent(ResourceActions::UPDATE, $configuration, $resource);
+        $this->manager->flush();
+        $this->eventDispatcher->dispatchPostEvent(ResourceActions::UPDATE, $configuration, $resource);
 
-            return $this->handleView($this->view($resource, 204));
+        if (!$configuration->isHtmlRequest()) {
+            return $this->viewHandler->handle($configuration, View::create($resource, 204));
         }
 
-        return $this->redirectHandler->redirectToReferer();
-    }
+        $this->flashHelper->addSuccessFlash($configuration, $enabled ? 'enable' : 'disable', $resource);
 
-    /**
-     * @return object
-     */
-    public function createNew()
-    {
-        return $this->resourceResolver->createResource($this->getFactory(), 'createNew');
+        return $this->redirectHandler->redirectToIndex($configuration, $resource);
     }
 
     /**
-     * @param object|null $resource
-     * @param array       $options
+     * @param Request $request
      *
-     * @return FormInterface
+     * @return Response
      */
-    public function getForm($resource = null, array $options = array())
+    public function moveUpAction(Request $request)
     {
-        $type = $this->config->getFormType();
-
-        if (strpos($type, '\\') !== false) { // full class name specified
-            $type = new $type();
-        } elseif (!$this->get('form.registry')->hasType($type)) { // form alias is not registered
```

Comment:
this is a regression, you have removed this, which means form types registered as services no longer work.